bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/handler/ModbusPollerThingHandlerImpl.java

```diff
@@ -296,6 +296,11 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public synchronized void initialize() {
+        if (this.getThing().getStatus().equals(ThingStatus.ONLINE)) {
```

Comment:
I wonder if this is the correct way to do it. @openhab/core-maintainers, I would expect that after `dispose()` was called, the thing-status should not be `ONLINE`, `OFFLINE` or `UNKNOWN` but `UNINITIALIZED` or something like that. Since `BaseThingHandler.dispose()` does not contain any code, the thing status stays unchanged even if `dispose()` has been called and `initialize()` is called again.

Reply:
Thank you @j-n-k for the review! 

I am not certain of the details so I simply followed what was previously done in the related PR.

Happy to accomodate the PR if there is a better way to do it of course.

 https://github.com/openhab/openhab-addons/pull/5244#issuecomment-476338666

Reply:
I still feel that this should be handled differently and that the bug is in the implementation in the `BaseThingHandler` that doesn't ensure that a disposed thing has a state that is not `ONLINE` but the bug needs to be fixed.